astropy/stats/lombscargle/tests/test_lombscargle.py

```diff
@@ -0,0 +1,161 @@
+import pytest
+import numpy as np
+from numpy.testing import assert_allclose, assert_equal
+
+from astropy import units
+
+from .. import LombScargle
+from ..implementations import lombscargle_slow, lombscargle
+
+
+@pytest.fixture
+def data(N=100, period=1, theta=[10, 2, 3], dy=1, rseed=0):
+    """Generate some data for testing"""
+    rng = np.random.RandomState(rseed)
+    t = 20 * period * rng.rand(N)
+    omega = 2 * np.pi / period
+    y = theta[0] + theta[1] * np.sin(omega * t) + theta[2] * np.cos(omega * t)
+    dy = dy * (0.5 + rng.rand(N))
+    y += dy * rng.randn(N)
+
+    return t, y, dy
+
+
+@pytest.mark.parametrize('method', LombScargle.available_methods)
+@pytest.mark.parametrize('shape', [(), (1,), (2,), (3,), (2, 3)])
+def test_output_shapes(method, shape, data):
+    t, y, dy = data
+    freq = np.asarray(np.random.rand(*shape))
+    freq.flat = np.arange(1, freq.size + 1)
+    PLS = lombscargle(t, y, frequency=freq,
+                      fit_bias=False, method=method)
+    assert_equal(PLS.shape, shape)
+
+
+@pytest.mark.parametrize('method', LombScargle.available_methods)
+@pytest.mark.parametrize('t_unit', [units.second, units.day])
+@pytest.mark.parametrize('frequency_unit', [units.Hz, 1. / units.second])
+@pytest.mark.parametrize('y_unit', [units.mag, units.jansky])
+def test_units_match(method, t_unit, frequency_unit, y_unit, data):
+    t, y, dy = data
+    dy = dy.mean()  # scipy only supports constant errors
+
+    t = t * t_unit
+    y = y * y_unit
+    dy = dy * y_unit
+    frequency = np.linspace(0.5, 1.5, 10) * frequency_unit
+    PLS = lombscargle(t, y, frequency=frequency,
+                      fit_bias=False, method=method)
+    assert_equal(PLS.unit, units.dimensionless_unscaled)
```

Comment:
That's one huge misconception: the periodogram does **not** reflect the amplitude of the Fourier fit.